Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -59,15 +110,91 @@ private void OpenUserInterface(EntityUid user, EntityUid analyzer)
+        /// </summary>
+        private void StopAnalyzingEntity(EntityUid uid, EntityUid healthAnalyzerUid, HealthAnalyzerComponent? healthAnalyzerComponent = null)
+        {
+            if (!TryComp<HealthBeingAnalyzedComponent>(uid, out var healthBeingAnalyzedComponent))
+                return;
+
+            //If there is more than 1 analyzer currently monitoring this entity, just remove from the list
+            if (healthBeingAnalyzedComponent.ActiveAnalyzers.Count > 1)
+            {
+                healthBeingAnalyzedComponent.ActiveAnalyzers.Remove(healthAnalyzerUid);
+            }
+            else
+            {
+                //If we are the last, remove the component
+                RemComp<HealthBeingAnalyzedComponent>(uid);
+            }
+
+
+            //Unlink the analyzer if it still exists
+            if (Resolve<HealthAnalyzerComponent>(healthAnalyzerUid, ref healthAnalyzerComponent, false))
+                healthAnalyzerComponent.ScannedEntity = null;
         }
 
-        public void UpdateScannedUser(EntityUid uid, EntityUid user, EntityUid? target, HealthAnalyzerComponent? healthAnalyzer)
+        private void OnDamageChanged(EntityUid damagedEntityUid, HealthBeingAnalyzedComponent component, DamageChangedEvent args)
         {
-            if (!Resolve(uid, ref healthAnalyzer))
+
+            //On the off chance that somehow this component is left behing with no entries - remove it
+            //Dont want this eating cpu cycles for no reason
+            if (component.ActiveAnalyzers.Count == 0)
+            {
+                RemComp<HealthBeingAnalyzedComponent>(damagedEntityUid);
                 return;
+            }
 
-            if (target == null || !_uiSystem.TryGetUi(uid, HealthAnalyzerUiKey.Key, out var ui))
+            foreach (var healthAnalyzerUid in component.ActiveAnalyzers)
+            {
+                //Strangeness catchall
+                //Should catch admin deleting the health analyzer, or something thats not an analyser ending up in the list
+                if (LifeStage(healthAnalyzerUid) >= EntityLifeStage.Terminating || !TryComp<HealthAnalyzerComponent>(healthAnalyzerUid, out var healthAnalyzerComp))
+                {
+                    StopAnalyzingEntity(damagedEntityUid, healthAnalyzerUid);
+                    continue;
+                }
+
+                //Get distance between health analyzer and the scanned entity
+                var scannedEntityPosition = _transformSystem.GetMapCoordinates(damagedEntityUid);
+                var healthAnalyserPosition = _transformSystem.GetMapCoordinates(healthAnalyzerUid);
+                var distance = (scannedEntityPosition.Position - healthAnalyserPosition.Position).Length();
+
+                if (scannedEntityPosition.MapId != healthAnalyserPosition.MapId ||
+                    distance > healthAnalyzerComp.MaxScanRange)
+                {
```

Comment:
You shouldn't be doing manual checks by hand, especially inside of a damage listener of all things.

Reply:
Removed the damage listener all together and replaced it with Update() and a 1hz update rate.

Reply:
Now using EntityCoordinates.InRange